### DISCUSSION TITLE
Triggering Backend Functions from the Admin

### DISCUSSION THREAD

**Discussion by andrashee** [socialincome-san/public#122] [Archived Discussions]
For some use cases, we would like to trigger a backend function from the Admin interface. E.g.

- Sending donation confirmations end of the year
- Creating the orange money payment lists
- ...

What do you think of the following proposal:
- we implement this functionalities in the backend as firebase functions. We expose them through a) [callable functions](https://firebase.google.com/docs/functions/callable) or b) just plain webhooks.
- in the callable functions, we automatically get the auth context and then can evaluate what role the user has and act accordingly.

Wdyt @mkue, @bojanmartin?

**Comment by mkue** [comment]
I think we should go for the first option: callable functions.

I already tested it :) See #124, I added a button in to the toolbar in FireCMS that triggers a function. Locally, this works like a charm.

**Comment by mkue** [comment]
It's also deployed here now: https://admin-social-income-prod--pr124-restructure-function-9s8cjzz7.web.app/

But there is a CORS policy issue😤

**Reply by mkue** [reply]
It seems to work in production though😅 Yay🎉

For testing, we should allow a `admin-social-income-prod.+\.web.app$` CORS regex somehow.
